index.js

```diff
@@ -96,43 +96,79 @@ original error: ${e}`);
   }
 }
 
-async function exportPodcasts(podcastsDBData) {
+
+async function buildPodcastDict(fileName, cacheFilesPath, podcastsDBData) {
+  const uuid = fileName.replace(".mp3", "");
+  const dbMeta = podcastsDBData.find((m) => m.zuuid === uuid);
+  const path = `${cacheFilesPath}/${fileName}`;
+  const exportBase = dbMeta?.zcleanedtitle // 1. from apple podcast database
+        ?? (await getMP3MetaTitle(path)) // 2. from mp3 meta data
+        ?? uuid; // 3. fallback to unreadable uuid
+  const podcastName = dbMeta?.zpodcast.replaceAll('/', '_');
+  const exportFileName = sanitize(exportBase.substr(0, fileNameMaxLength));
+  const date = dbMeta?.date
+
+  const ret = {
+    podcastName,
+    date,
+    fileName,
+    path,
+    uuid,
+    exportFileName: `${exportFileName}.mp3`
+  };
+  return ret;
+}
+
+
+function filterPodcasts(podcasts, filepatterns = []) {
+  if (filepatterns.length == 0) {
+    return podcasts;
+  }
+
+  function matchesAny(s) {
+    return filepatterns.some((p) => { return s.indexOf(p) != -1 })
+  }
```

Comment:
in that case two variables with one letter name don't look very readable, then `includes` looks cleaner and I would also handle upper and lower cases , 
I propose something like 
```suggestion
  function matchesAny(fileOrDir) {
    return filepatterns
      .map(pattern => pattern.toLowerCase())
      .some((pattern) => fileOrDir.toLowerCase().includes(pattern) )
  }
```

Reply:
Looks good to me.

Reply:
Added.